template/template.go

```diff
@@ -175,19 +180,19 @@ func (t *Template) Execute(w io.Writer, input map[string]any) error {
 	for k, v := range input {
 		var f Func
 
-		def := NewResolvedValue(v, WithFormat(func() string { return defaultFormat(v) }))
+		r := NewResolvedValue(v, WithFormat(func() string { return defaultFormat(v) }))
 
 		switch v.(type) {
 		default:
-			executer.variables[k] = def
+			executer.variables[k] = r
```

Comment:
```suggestion
		switch v.(type) {
		default:
			executer.variables[k] = NewResolvedValue(v, WithFormat(func() string { return defaultFormat(v) }))
```

Reply:
I'm using the `r` also as an input to the registry func on line 195.

Reply:
Are you sure? :) It is a new declaration, not an assignment to the previous _r_ variable.

Reply:
Oh no, it bit me again..